### PR TITLE
Allow having multiple locks with different keys

### PR DESCRIPTION
**Pull request by kasbah**
This lets you do:

```js
await lock.acquireAsync({key: 'foo'})
await lock.acquireAsync({key: 'bar'})
await lock.release({key: 'foo'})
...
```

### CODE REVIEW

File: src/AwaitLock.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
   #waitingResolvers: Set<() => void> = new Set();
 
   /**
    * Whether the lock is currently acquired or not. Accessing this property does not affect the
    * status of the lock.
    */
   get acquired(): boolean {
-    return this.#acquired;
+    return this.#acquired[defaultKey];
   }
```
Comment (kasbah) [inline src/AwaitLock.ts:16]: Left it like this for maximum backwards compatibility but this should probably be changed to a method to have the optional key. 

```js
getAcquired(key?: string)
```

**Comment by ide** [comment]
It looks like this use case could be addressed in user space by creating a map with multiple locks. IME, lock libraries provide single locks.

**Comment by kasbah** [comment]
That's true, in fact I've done it that way. I just thought it would be neater if this library provides the mechanism. Here's my implementation in case it's useful to someone.

```ts
import AwaitLock from 'await-lock'

const locks = {}

async function acquireLock(key: string): Promise<void> {
  if (!Object.keys(locks).includes(key)) {
    locks[key] = new AwaitLock()
  }
  await locks[key].acquireAsync()
}

async function releaseLock(key: string): Promise<void> {
  if (!Object.keys(locks).includes(key)) {
    console.warn('Releasing unknown lock:', key)
    return
  }
  await locks[key].release()
}
```

Are you set on not adding this functionality to this lib then?

**Comment by ide** [comment]
I'd rather keep the API surface area small and leave behavior that can be reasonably implemented in user space out of the library, especially APIs that aren't common to several established lock APIs (Java, Python, POSIX, etc.).

**Pull request closed without merging**